src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionTelemetry.cs

```diff
@@ -158,19 +159,38 @@ internal static string GetHostnameFromStateObject(object hostNameOrAddress)
     /// </summary>
     internal readonly struct NameResolutionActivity
     {
+        [FeatureSwitchDefinition("System.Diagnostics.ActivitySource.IsSupported")]
+        private static bool IsActivitySourceSupported { get; } = InitializeIsActivitySourceSupported();
+        private static bool InitializeIsActivitySourceSupported() => AppContext.TryGetSwitch("System.Diagnostics.ActivitySource.IsSupported", out bool isSupported) ? isSupported : true;
+
         private const string ActivitySourceName = "Experimental.System.Net.NameResolution";
         private const string ActivityName = ActivitySourceName + ".DnsLookup";
-        private static readonly ActivitySource s_activitySource = new ActivitySource(ActivitySourceName);
+        private static readonly ActivitySource? s_activitySource;
```

Comment:
```suggestion
        private static readonly ActivitySource? s_activitySource = IsActivitySourceSupported ? new ActivitySource(ActivitySourceName) : null;
```
Instead of the explicit constructor and warning disable?

Reply:
It doesn't trim this way. It linker would leave the field in place assigned to null. But the class would survive.

Reply:
But maybe I would have to give up trimming ActivitySource class anyway

Reply:
I am not able to reproduce the trimming issue with a small standalone test. The `? :` expression works as good as the `if else` as far as I can tell.

> I would have to give up trimming ActivitySource class anyway

You should be able to get ActivirtySource instance constructor and instance methods trimmed or replaced with dummy bodies. I do not think you can make the ActivitySource to go away completely. I expect static constructor and some of the static methods will be kelp.

Reply:
There are class references also on too many APIs and so trimming this class is moot.